### PR TITLE
feat(ext/net): support NAPTR records

### PR DESCRIPTION
**Pull request by cmorten**
Add `NAPTR` record type support for `Deno.resolveDns()`.

Very new to Rust, so guidance much appreciated!

### CODE REVIEW

**Comment by cmorten** [comment]
Real world example:

```ts
// trial.ts
console.log(
  await Deno.resolveDns("4.4.2.2.3.3.5.6.8.1.4.4.e164.arpa", "NAPTR")
); // expect resolution

await Deno.resolveDns("google.com", "NAPTR"); // expect NotFound
```

```console
$ ./target/debug/deno run -A --unstable ./trial.ts
Check file:///Users/craigmorten/git/cmorten/deno/trial.ts
[
  {
    order: 100,
    preference: 10,
    flags: "u",
    services: "E2U+sip",
    regexp: "!^\\+441865332(.*)$!sip:\\1@nominet.org.uk!",
    replacement: "."
  },
  {
    order: 100,
    preference: 20,
    flags: "u",
    services: "E2U+pstn:tel",
    regexp: "!^(.*)$!tel:\\1!",
    replacement: "."
  }
]
error: Uncaught (in promise) NotFound: no record found for name: google.com.lan. type: NAPTR class: IN
await Deno.resolveDns("google.com", "NAPTR");
^
    at async file:///Users/craigmorten/git/cmorten/deno/trial.ts:5:1
```

File: cli/dts/lib.deno.ns.d.ts [hunk 1]
```diff
@@ -3044,5 +3061,12 @@ declare namespace Deno {
     query: string,
     recordType: RecordType,
     options?: ResolveDnsOptions,
-  ): Promise<string[] | MXRecord[] | SRVRecord[] | string[][]>;
+  ): Promise<
+    | string[]
+    | MXRecord[]
+    | NAPTRRecord[]
+    | SOARecord[]
```
Comment (cmorten) [inline cli/dts/lib.deno.ns.d.ts:3068]: Seems was missed in #14534

File: cli/diagnostics.rs [hunk 2]
```diff
@@ -29,6 +29,7 @@ const UNSTABLE_DENO_PROPS: &[&str] = &[
   "Location",
   "MXRecord",
   "Metrics",
+  "NAPTRRecord",
```
Comment (bartlomieju) [inline cli/diagnostics.rs:32]: Hmmm, this shouldn't be needed... `Deno.resolveDns()` is not stable API. We should remove all of the types related to that API from this list

**Review by bartlomieju — APPROVED** [review]
This looks good, no guidance needed :)

File: cli/diagnostics.rs (hunk 2 above)
Comment (bartlomieju) [reply cli/diagnostics.rs:32]: Opened https://github.com/denoland/deno/pull/14618

Comment (cmorten) [reply cli/diagnostics.rs]: Ah oops! Good catch 😄

**Pull request merged**